src/main/kotlin/com/github/igorperikov/mightywatcher/service/OutputService.kt

```diff
@@ -0,0 +1,85 @@
+package com.github.igorperikov.mightywatcher.service
+
+import com.github.igorperikov.mightywatcher.Issues
+import com.github.igorperikov.mightywatcher.entity.NamedTimestamp
+import j2html.TagCreator.*
+import j2html.attributes.Attr
+import j2html.tags.ContainerTag
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+const val PDF_OUTPUT_TYPE = "PDF"
+const val HTML_OUTPUT_TYPE = "HTML"
+const val CONSOLE_OUTPUT_TYPE = "CONSOLE"
+
+
+class OutputService(
+        private val type: String
+) {
+    companion object {
+        const val HTML_PATH_FORMAT = "might-watch-report-%s.html"
+        @JvmField
+        val FORMATTER: DateTimeFormatter = DateTimeFormatter.ofPattern("dd-MMMM-yyyy")
+    }
+
+    private val log = LoggerFactory.getLogger(this.javaClass)
+
+    fun getResults(issues: LinkedHashMap<NamedTimestamp, Issues>) {
```

Comment:
It feels like this class responsible for 2 different things - it converts data in proper format AND prints it. Separating them would be easier to test and to use. What do you think?

Reply:
That's the problem I was talking for a while, i do not know how to solve it properly with existing data structure

Reply:
One service can return list of some sort of entities, e.g. (naming is bad atm)
```kotlin
class TransformingService {
  fun transform(issues: LinkedHashMap<NamedTimestamp, Issues>): List<ResultIssue> { ... }
}

interface ResultLine {}
class IssueLine(repoName: String, title: String, htmlUrl: String) : ResultLine { ... }
class TimestampLine(name: String) : ResultLine { ... }
```

Whilst, another one will do the thing to them
```kotlin
interface PrintingService {
  fun print(resultLines: List<ResultLine>)
}
class ConsolePrintingService : PrintingService { ... }
```

briefly that's what comes up in my mind